lib/pure/net.nim

```diff
@@ -1542,7 +1542,7 @@ proc readLine*(socket: Socket, line: var string, timeout = -1,
     var c: char
     discard waitFor(socket, waited, timeout, 1, "readLine")
     var n = recv(socket, addr(c), 1)
-    if n < 0: raiseSockError()
+    if n < 0: raiseSockError(); return
```

Comment:
I think you should put this in the `raiseSockError` template

Reply:
I'm pretty new with Nim. If I put it into the template, the `return` will exit `raiseSockError()` and not the `readLine` function? Or is it somehow propagated?

Reply:
Because the template is "dirty" the call to the template gets replaced with its contents